jumanji/environments/routing/connector/env.py

```diff
@@ -180,31 +177,26 @@ def step(
             grid=grid, step_count=state.step_count + 1, agents=agents, key=state.key
         )
 
-        # Construct timestep: get observations, rewards, discounts
-        grids = self._obs_from_grid(grid)
+        # Construct timestep: get rewards, discounts
         reward = self._reward_fn(state, action, new_state)
         action_mask = jax.vmap(self._get_action_mask, (0, None))(agents, grid)
         observation = Observation(
-            grid=grids, action_mask=action_mask, step_count=new_state.step_count
+            grid=grid, action_mask=action_mask, step_count=new_state.step_count
         )
 
-        dones = jax.vmap(connected_or_blocked)(agents, action_mask)
-        discount = jnp.asarray(jnp.logical_not(dones), dtype=float)
+        dones = jnp.all(jax.vmap(connected_or_blocked)(agents, action_mask))
```

Comment:
Should this be singular?
```suggestion
        done = jnp.all(jax.vmap(connected_or_blocked)(agents, action_mask))
```